install/install.sh

```diff
@@ -45,328 +45,362 @@ error() {
-    ;;
-
-  *)
-    error "Unknown option: $1"
-    exit 1
-    ;;
+    -p | --platform)
+      platform="$2"
+      shift 2
+      ;;
+    -b | --bin-dir)
+      BIN_DIR="$2"
+      shift 2
+      ;;
+    -a | --arch)
+      arch="$2"
+      shift 2
+      ;;
+    -B | --base-url)
+      base_url="$2"
+      shift 2
+      ;;
+
+    -V | --verbose)
+      verbose=1
+      shift 1
+      ;;
+    -f | -y | --force | --yes)
+      force=1
+      shift 1
+      ;;
+
+    -p=* | --platform=*)
+      platform="${1#*=}"
+      shift 1
+      ;;
+    -b=* | --bin-dir=*)
+      BIN_DIR="${1#*=}"
+      shift 1
+      ;;
+    -a=* | --arch=*)
+      arch="${1#*=}"
+      shift 1
+      ;;
+    -B=* | --base-url=*)
+      base_url="${1#*=}"
+      shift 1
+      ;;
+    -V=* | --verbose=*)
+      verbose="${1#*=}"
+      shift 1
+      ;;
+    -f=* | -y=* | --force=* | --yes=*)
+      force="${1#*=}"
+      shift 1
+      ;;
+
+    *)
+      error "Unknown option: $1"
+      exit 1
+      ;;
   esac
 done
 
-if [ "${ARCH}" = "i386" ]; then
-  error "i386 builds are not yet available for Starship\n"
-  info "If you would like to see a build for your configuration,"
-  info "please create an issue requesting a build for ${MAGENTA}${ARCH}-${PLATFORM}${NO_COLOR}:"
-  info "${BOLD}${UNDERLINE}https://github.com/starship/starship/issues/new/${NO_COLOR}\n"
-  exit 1
-fi
+arch_check
+platform_check
+bin_dir_check
 
 printf "  %s\n" "${UNDERLINE}Configuration${NO_COLOR}"
 info "${BOLD}Bin directory${NO_COLOR}: ${GREEN}${BIN_DIR}${NO_COLOR}"
-info "${BOLD}Platform${NO_COLOR}:      ${GREEN}${PLATFORM}${NO_COLOR}"
-info "${BOLD}Arch${NO_COLOR}:          ${GREEN}${ARCH}${NO_COLOR}"
+info "${BOLD}Platform${NO_COLOR}:      ${GREEN}${platform}${NO_COLOR}"
+info "${BOLD}Arch${NO_COLOR}:          ${GREEN}${arch}${NO_COLOR}"
+printf "\n"
 
-# non-empty VERBOSE enables verbose untarring
-if [ -n "${VERBOSE-}" ]; then
-  VERBOSE=v
+# non-empty `verbose` enables verbose untarring
+if [ -n "$verbose" ]; then
   info "${BOLD}Verbose${NO_COLOR}: yes"
-else
-  VERBOSE=
+  printf "\n"
 fi
 
-echo
+tmp=$(mktmpdir "starship")
+case $platform in
+  pc-windows-msvc)
+    bin="$tmp/starship.exe"
+    ext="zip"
+    ;;
+  *)
+    bin="$tmp/starship"
+    ext="tar.gz"
+    ;;
+esac
+archive="$tmp/starship.$ext"
+
+base_url=${base_url:-"https://github.com/starship/starship/releases"}
+url="$base_url/latest/download/starship-$arch-$platform.$ext"
 
-EXT=tar.gz
-if [ "${PLATFORM}" = "pc-windows-msvc" ]; then
-  EXT=zip
+info "Tarball URL: ${UNDERLINE}${BLUE}${url}${NO_COLOR}"
+if [ -z "$force" ]; then
+  confirm "Install Starship ${GREEN}latest${NO_COLOR} to ${BOLD}${GREEN}${BIN_DIR}${NO_COLOR}?"
 fi
+path_check
 
-URL="${BASE_URL}/latest/download/starship-${ARCH}-${PLATFORM}.${EXT}"
-info "Tarball URL: ${UNDERLINE}${BLUE}${URL}${NO_COLOR}"
-confirm "Install Starship ${GREEN}latest${NO_COLOR} to ${BOLD}${GREEN}${BIN_DIR}${NO_COLOR}?"
-check_bin_dir "${BIN_DIR}"
+if [ -w "$BIN_DIR" ]; then
```

Comment:
I'll need to test this again, but I thought that `-w` on Windows doesn't actually tell you if the directory is writeable or not.

Reply:
I wanted to ask you about that. It really should, previous workaround with `type` was meant to be used inside batch scripts. This is something that bash should provide to us.

Reply:
Just to confirm: for some reason or other, `-w` on Git Bash in windows incorrectly believes that you can write into system directories. Maybe we should bring back the old test-writeability function that actually tries to write a file?

Reply:
> Just to confirm: for some reason or other, `-w` on Git Bash in windows incorrectly believes that you can write into system directories. Maybe we should bring back the old test-writeability function that actually tries to write a file?

Windows, again 😫  Ok, it seems we need to bring it back. I'll update this later on 👍